frame/dvm/src/lib.rs

```diff
@@ -220,11 +247,20 @@ decl_module! {
 		}
 
 		fn on_finalize(_block_number: T::BlockNumber) {
-			<Module<T>>::store_block();
+			<Module<T>>::store_block(
+				dp_consensus::find_pre_log(&frame_system::Module::<T>::digest()).is_err(),
+			);
 		}
 
 		fn on_initialize(_block_number: T::BlockNumber) -> Weight {
 			Pending::kill();
+			if let Ok(log) = dp_consensus::find_pre_log(&frame_system::Module::<T>::digest()) {
+				let PreLog::Block(block) = log;
+
+				for transaction in block.transactions {
+					let _ = Call::<T>::transact(transaction).dispatch_bypass_filter(RawOrigin::None.into());
```

Comment:
This looks like it will always fail due to the check in dispatch call, as we discussed.

Reply:
This question is not answered in the upstream library at the moment. Let me create an issue to track, and do not discuss it too much in this pr, it's already huge enough.